src/sql/engine/px/ob_dfo_scheduler.cpp

```diff
@@ -1122,12 +1130,23 @@ int ObParallelDfoScheduler::dispatch_sqc(ObExecContext &exec_ctx,
       LOG_WARN("session is NULL", K(ret));
     }
   }
-  if (OB_SUCC(ret) && nullptr != dfo.parent() && dfo.parent()->is_root_dfo()) {
+  if (OB_SUCC(ret)) {
     ARRAY_FOREACH(sqcs, idx) {
       ObPxSqcMeta &sqc = *sqcs.at(idx);
-      sqc.set_adjoining_root_dfo(true);
+      // 设置sqc的record_data_字段
+      ObQcRecordData& record_data = sqc.get_record_data();
```

Comment:
same question as above.

Reply:
> same question as above.

I have put it to ObPXServerAddrUtil::alloc_by_XXX_distribution when ObPxSqcMeta is initializing